### PR TITLE
FIX: support Qt 5.15

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

 - QPainter.drawLine is now picky about float vs int
 - be more careful that we close the Qpainter out

I may need to check the exact version we need this for more carefully.  May be a pyqt, not Qt change.

### CODE REVIEW

File: lib/matplotlib/backends/backend_qt5.py [hunk 1]
```diff
@@ -486,7 +486,7 @@ def drawRectangle(self, rect):
         # Draw the zoom rectangle to the QPainter.  _draw_rect_callback needs
         # to be called at the end of paintEvent.
         if rect is not None:
-            x0, y0, w, h = [pt / self._dpi_ratio for pt in rect]
+            x0, y0, w, h = map(int, [pt / self._dpi_ratio for pt in rect])
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_qt5.py:489]: You could just use `int` directly?
```suggestion
            x0, y0, w, h = [int(pt / self._dpi_ratio) for pt in rect]
```

Comment (tacaswell) [reply lib/matplotlib/backends/backend_qt5.py]: fair, `map` is what popped into my head first..

**Comment by QuLogic** [comment]
Strange, the `.end()` docs say you don't need to call it as it's done in the destructor, but many examples still seem to show it.

File: lib/matplotlib/backends/backend_qt5.py [hunk 2]
```diff
@@ -486,7 +486,7 @@ def drawRectangle(self, rect):
         # Draw the zoom rectangle to the QPainter.  _draw_rect_callback needs
         # to be called at the end of paintEvent.
         if rect is not None:
-            x0, y0, w, h = [pt / self._dpi_ratio for pt in rect]
+            x0, y0, w, h = [int(pt / self._dpi_ratio) for pt in rect]
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_qt5.py:489]: Or round? c.f. https://github.com/matplotlib/matplotlib/pull/15656#discussion_r344543106

I'm not sure myself. I think I tested 15656 with round but it caused some test failures. Did not have time to investigate.

Maybe `int()` is good enough to fix this if we don't have the capacity to work out `round` vs. `int`.

Comment (tacaswell) [reply lib/matplotlib/backends/backend_qt5.py:489]: We have looked at moving more things to `round` in other places (https://github.com/matplotlib/matplotlib/pull/8265) and it also caused failures.  I am inclined to stay with int here as I suspect that is what is used to be doing under the hood (it is just complaining about in now).

On the other hand, this is in drawing the zoom box so it probably doesn't really matter?

**Comment by QuLogic** [comment]
> I may need to check the exact version we need this for more carefully. May be a pyqt, not Qt change.

Are you okay to merge now, or still wait?

**Comment by tacaswell** [comment]
I'm OK to merge, the version checking would only be to mkae the comit message more accurate.

**Pull request merged**